h/tasks/indexer.py

```diff
@@ -52,6 +53,10 @@ def report_job_queue_metrics():
 
     newrelic.agent.record_custom_metrics(
         [
+            (
+                "Custom/SyncAnnotations/Queue/Length",
+                celery.request.db.query(Job).count(),
```

Comment:
Since this is blindly counting all jobs, rather than counting only `"sync_annotation"` jobs, it's not really specific to / doesn't really belong in `services.search_index.Queue`, which is supposed to be specific to `"sync_annotation"` jobs. And we don't need to encapsulate `models.Job` within `search_index.Queue`, it was always the plan that multiple different services and other things could build on this model for different purposes.

Also I think it might be trying too hard to try to make `Queue.count()` be able to do this, when SQLAlchemy already provides a trivial way to do it.

Reply:
I've gone with `Custom/Job/Queue/Length`, 'cos I think I do just want a simple measure of the total length of the job queue, completely unfiltered. This isn't for any particular purpose (e.g. there's no alarm I'm planning based on this), I just think that we should be able to have one graph on the job queue dashboard that just shows the total unfiltered length of the queue.

I think the existing `"Custom/SyncAnnotations/Queue/API/Length"` should actually count the unscheduled jobs (there's no reason to filter those out of the count) but I think that's a separate issue

Reply:
Separate PR to add not-yet-scheduled jobs to the `"Custom/SyncAnnotations/Queue/API/Length"` metric: https://github.com/hypothesis/h/pull/6392